packages/vscode-extension/src/webview/components/InspectDataMenu.tsx

```diff
@@ -37,40 +77,57 @@ export function InspectDataMenu({
   })();
 
   const filteredData = inspectStack.filter((item) => !item.hide);
+  const inspectItems =
+    shouldShowAll || filteredData.length === MAX_INSPECT_ITEMS + 1
+      ? filteredData
+      : filteredData.slice(0, MAX_INSPECT_ITEMS);
+  const inspectMenuAlign = inspectLocation.x <= window.innerWidth / 2 ? "start" : "end";
+  const isOverMaxItems = filteredData.length > MAX_INSPECT_ITEMS + 1;
 
   return (
-    <div style={{ left: inspectLocation.x, top: inspectLocation.y, position: "absolute" }}>
-      <DropdownMenu.Root
-        defaultOpen={true}
-        open={true}
-        onOpenChange={(open) => {
-          if (!open) {
-            onCancel();
-          }
-        }}>
-        <DropdownMenu.Trigger />
-        <DropdownMenu.Portal>
-          <DropdownMenu.Content className="inspect-data-menu-content">
-            <DropdownMenu.Label className="inspect-data-menu-label">
-              {displayDimensionsText}
-            </DropdownMenu.Label>
-            {filteredData.map((item, index) => {
-              // extract file name from path:
-              const fileName = item.source.fileName.split("/").pop();
-              return (
-                <DropdownMenu.Item
-                  className="inspect-data-menu-item"
-                  key={index}
-                  onSelect={() => onSelected(item)}
-                  onMouseEnter={() => onHover(item)}>
-                  <code>{`<${item.componentName}>`}</code>
-                  <div className="right-slot">{`${fileName}:${item.source.line0Based + 1}`}</div>
-                </DropdownMenu.Item>
-              );
-            })}
-          </DropdownMenu.Content>
-        </DropdownMenu.Portal>
-      </DropdownMenu.Root>
-    </div>
+    <DropdownMenu.Root
+      defaultOpen={true}
+      open={true}
+      onOpenChange={(open) => {
+        if (!open) {
+          onCancel();
+        }
+      }}>
+      <DropdownMenu.Trigger asChild>
+        <span
+          style={{
+            position: "absolute",
+            left: inspectLocation.x,
+            top: inspectLocation.y,
+            opacity: 0,
+          }}
+        />
+      </DropdownMenu.Trigger>
+      <DropdownMenu.Portal>
+        <DropdownMenu.Content
+          className="inspect-data-menu-content"
+          sideOffset={5}
+          align={inspectMenuAlign}
+          collisionPadding={5}>
+          <DropdownMenu.Label className="inspect-data-menu-label">
+            {displayDimensionsText}
+          </DropdownMenu.Label>
+          {inspectItems.map((item) => (
+            <InspectItem item={item} onSelected={onSelected} onHover={onHover} />
+          ))}
+          {isOverMaxItems && !shouldShowAll && (
+            <DropdownMenu.Item
+              className="inspect-data-menu-item show-all"
+              key={"show-all"}
```

Comment:
I think now the `key` is unnecessary as the elements here differ

Reply:
Its needed to avoid warrning `[WEBVIEW LOG] Warning: Each child in a list should have a unique "key" prop.%s%s` in logs.

Reply:
Interesting